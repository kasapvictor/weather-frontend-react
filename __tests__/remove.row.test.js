import * as fs from 'fs';
import * as path from 'path';
import {dirname} from 'path';
import { fileURLToPath } from 'url';
import { queryByAttribute } from '@testing-library/dom';
import '@testing-library/jest-dom';
import testingLibraryUserEvent from '@testing-library/user-event';

const userEvent = testingLibraryUserEvent.default;
const fileName = fileURLToPath(import.meta.url);
const dirName = dirname(fileName);
const getFixturePath = (filename) => path.join(dirName, '..', '__fixtures__', filename);
const getContent = (file) => fs.readFileSync((file), 'utf-8');

test('Check Remove Row', () => {
	const div = document.createElement('div')
	const beforePath = getFixturePath('remove.row.html');
	div.innerHTML = getContent(beforePath);
	
	const button = div.querySelector('[data-testid="524894"]');
	const row = div.querySelector('[data-row-city="524894"]');
	
	button.addEventListener('click', () => {
		row.remove();
	});
	userEvent.click(button);
	
	expect(queryByAttribute('data-row-city', div, '524894')).toBeNull();
	expect(div).toMatchSnapshot()
});
