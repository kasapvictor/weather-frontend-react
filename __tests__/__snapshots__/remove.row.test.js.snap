// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Check Remove Row 1`] = `
<div>
  <div
    class="table__body"
    id="table-body"
  >
    
    
    
    
    <div
      class="table__row"
      data-row-city="1496747"
    >
      
        
      <button
        class="table__row-remove"
        data-remove-city="1496747"
        data-testid="1496747"
      >
        ×
      </button>
      
    
    </div>
    
    
    <div
      class="table__row"
      data-row-city="504341"
    >
      
        
      <button
        class="table__row-remove"
        data-remove-city="504341"
      >
        ×
      </button>
      
    
    </div>
    

  </div>
  

</div>
`;
